circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -43,13 +42,19 @@ trait CirceInstances extends JawnInstances {
   protected def jsonDecodeError: (Json, NonEmptyList[DecodingFailure]) => DecodeFailure =
     CirceInstances.defaultJsonDecodeError
 
-  def jsonDecoderIncremental[F[_]: Sync]: EntityDecoder[F, Json] =
+  def jsonDecoderIncremental[F[_]](implicit
+      C: Stream.Compiler[F, F],
+      A: ApplicativeThrow[F]): EntityDecoder[F, Json] =
```

Comment:
The only `F` that satisfies both is one which has `Sync`, so I don't think this helps. Also, it would break backwards compatibility

Reply:
There is also a `Stream.Compiler[Id, Id]`, which I think is what motivated @hamnis.